### PR TITLE
supported dubbo-go v3.0.2

### PR DESCRIPTION
**Pull request by dgqypl**

### CODE REVIEW

**Comment by dgqypl** [comment]
> Hi @dgqypl , please help to replace the [link](https://github.com/SkyAPM/go2sky-plugins/blob/master/Makefile#L30) with `https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh` :)

curl -L https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.45.2 -> curl -L https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.45.2
right?

File: dubbo-go/dubbo-go.go [hunk 1]
```diff
@@ -125,15 +125,18 @@ func (cf tracingFilter) Invoke(ctx context.Context, invoker protocol.Invoker, in
...
 
 		span.SetComponent(componentID)
 	} else {
 		// componentIDGo2SkyServer
 		span, ctx, _ = cf.tracer.CreateEntrySpan(ctx, operationName, func(key string) (string, error) {
-			return invocation.AttachmentsByKey(key, ""), nil
+			if attachment, ok := invocation.GetAttachment(key); ok {
+				return attachment, nil
+			}
+			return "", nil
```
[the remark this answers is not in the document]
Comment (dgqypl) [reply dubbo-go/dubbo-go.go:139]: see https://github.com/SkyAPM/go2sky-plugins/pull/49/commits/e0804e605b8fdb4c1fe5a87896fa26d52990a0d2 go.mod

[the remark this answers is not in the document]
Comment (wu-sheng) [reply dubbo-go/dubbo-go.go:139]: OK, so move from RC to GA release.

**Comment by arugal** [comment]
> > Hi @dgqypl , please help to replace the [link](https://github.com/SkyAPM/go2sky-plugins/blob/master/Makefile#L30) with `https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh` :)
> 
> 
> 
> curl -L https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.45.2 -> curl -L https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.45.2
> 
> right?

Yes

**Comment by AlexStocks** [comment]
@dgqypl thank you for your great work. pls attention that the current dubbogo version is v3.0.2,  pls upgrade dubbogo version to the latest version. thank you.

**Comment by dgqypl** [comment]
> @dgqypl thank you for your great work. pls attention that the current dubbogo version is v3.0.2,  pls upgrade dubbogo version to the latest version. thank you.

I'll upgrade that tomorrow 😴.

**Comment by dgqypl** [comment]
> @dgqypl thank you for your great work. pls attention that the current dubbogo version is v3.0.2, pls upgrade dubbogo version to the latest version. thank you.

already up to date

**Comment by AlexStocks** [comment]
> > @dgqypl thank you for your great work. pls attention that the current dubbogo version is v3.0.2,  pls upgrade dubbogo version to the latest version. thank you.
> 
> I'll upgrade that tomorrow 😴.

Please change this pr's title, thx.

**Comment by dgqypl** [comment]
> @dgqypl Also need to look at why [golang-lint fails](https://github.com/SkyAPM/go2sky-plugins/runs/7260515074?check_suite_focus=true), are you interested?
> 
> 
> 
> I was successful on go 1.18.3, which is probably related to the golang version

Actually, I don't know how to do right now, can you give me some tips, thx.

**Comment by dgqypl** [comment]
> > > @dgqypl thank you for your great work. pls attention that the current dubbogo version is v3.0.2,  pls upgrade dubbogo version to the latest version. thank you.
> 
> > 
> 
> > I'll upgrade that tomorrow 😴.
> 
> 
> 
> Please change this pr's title, thx.

changed

**Comment by AlexStocks** [comment]
> @dgqypl Also need to look at why [golang-lint fails](https://github.com/SkyAPM/go2sky-plugins/runs/7260515074?check_suite_focus=true), are you interested?
> 
> I was successful on go 1.18.3, which is probably related to the golang version

as a basic lib, the old version if perfect for many users. the older, the better.

**Comment by wu-sheng** [comment]
Dubbo go case seems running on go 1.14. Is 14 still working for this latest Dubbo version?

**Comment by wu-sheng** [comment]
Take a look at the GitHub control file.

**Comment by dgqypl** [comment]
dubbo-go v3.0.2's go.mod file use go 1.15

**Comment by wu-sheng** [comment]
I mean CI control file, I am not sure which part controls env initialization.

**Comment by arugal** [comment]
> 

https://github.com/SkyAPM/go2sky-plugins/blob/master/.github/workflows/ci.yaml#L42

@dgqypl  Please change the golang version here to 1.15

**Comment by arugal** [comment]
https://github.com/SkyAPM/go2sky-plugins/runs/7260518151?check_suite_focus=true

E2E test also failed, please check

**Comment by dgqypl** [comment]
> > 
> 
> 
> 
> https://github.com/SkyAPM/go2sky-plugins/blob/master/.github/workflows/ci.yaml#L42
> 
> 
> 
> @dgqypl  Please change the golang version here to 1.15

ok, I'll modify later.

**Comment by dgqypl** [comment]
> https://github.com/SkyAPM/go2sky-plugins/runs/7260518151?check_suite_focus=true
> 
> E2E test also failed, please check

right now left only this fails, how can I fix it?

**Comment by arugal** [comment]
> Details

https://github.com/SkyAPM/go2sky-plugins/runs/7264192077?check_suite_focus=true#step:4:248

e2e logs show that the connection between dubbo-go and zookeeper is abnormal

**Comment by dgqypl** [comment]
> > Details
> 
> https://github.com/SkyAPM/go2sky-plugins/runs/7264192077?check_suite_focus=true#step:4:248
> 
> e2e logs show that the connection between dubbo-go and zookeeper is abnormal

looks like that docker-compose.yml may be modify somewhere, or else?

**Comment by wu-sheng** [comment]
@dgqypl Could you fix the test?

**Comment by dgqypl** [comment]
> @dgqypl Could you fix the test?

I had already fix it right now.

**Comment by dgqypl** [comment]
from the latest container log, it shows:
```
2022-07-11T01:57:01.764Z	ERROR	config/provider_config.go:156	Dubbogo do not read service interface name with registeredTypeName = UserProvider.Please run go install github.com/dubbogo/tools/cmd/protoc-gen-go-triple@latest to update your protoc-gen-go-triple and re-generate your pb file again.If you are not using pb serialization, please set 'interface' field in service config.
```
I really don't know how to fix it. Is there anyone who can **directly** tell me how to do.

**Comment by wu-sheng** [comment]
This seems a dubbo go error? I think it is better to ask them.

**Comment by dgqypl** [comment]
@AlexStocks can you help me to fix it? thx

**Comment by zhaoyunxing92** [comment]
> @AlexStocks can you help me to fix it? thx

how should i contact you

**Comment by AlexStocks** [comment]
> @AlexStocks can you help me to fix it? thx

Pls join the dingding group 23331795 and contact with this group owner yuyu[me] and then let's have a discussion with @zhaoyunxing92 .

**Comment by dgqypl** [comment]
I had already fix CI error associated with dubbogo, so the only thing left is CI error with [expected.data.yml](https://github.com/SkyAPM/go2sky-plugins/blob/master/dubbo-go/test/expected.data.yml), I don't know how to modify this file and let the CI passed, anybody can help me to resolve this problem?

**Comment by wu-sheng** [comment]
This is an e2e test framework go2sky is using. 

https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/plugin-test/

This section is about the expected file, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/plugin-test/#expecteddatayaml

Generally, it is about what the traces you collected should be

**Comment by wu-sheng** [comment]
I think you forgot to revert GHA file changes.

**Comment by dgqypl** [comment]
> 

The latest commit is reverted , please try again the workflows, I'm tested CI passed in my forked repository.

**Comment by wu-sheng** [comment]
@arugal @AlexStocks Please take a look.

**Review by arugal — APPROVED** [review]
LGTM, Thank you for your contribution :)

**Review by AlexStocks — APPROVED** [review]
LGTM

**Pull request merged**